pkg/sync/sync.go

```diff
@@ -117,7 +118,8 @@ func NewTask(ctx context.Context, controllerAddress string) (*Task, error) {
 	}()
 
 	return &Task{
-		client: controllerClient,
+		client:     controllerClient,
+		volumeName: volumeName,
```

Comment:
Can we cache `volumeName` in `ControllerClient`?

Reply:
I think this is somewhat awkward because of the way we reference it (instead of grabbing it directly from the task we must access it as an exported field of the controller client every time we instantiate a replica client. However, I don't feel strongly about it. Made the change in the latest commit.